### PR TITLE
Remove experimental ES7 code, and use babel for webpack config

### PR DESCRIPTION
**Pull request by jdan**
Some brief cleanup before I dive into some more interesting cleanup! This changeset does the following: 

* removes the neat-but-suspicious ES7 function-bind code (`::`)
* removes the `babel-eslint` dependency
* moves babel config info `webpack.config` for consistency
* DRYs up the code to determine what function to transform JSX into
* converts `webpack.config.js` into ES6!

No functional changes. `npm run live-test` works the same as it did before.

@rileyjshaw PTAL

### CODE REVIEW

File: utils/webpack.config.babel.js [hunk 1]
```diff
@@ -43,15 +48,15 @@ module.exports = {
...
             }),
             {entryOnly: true}),
 
-        // Make the JSX pragma function "E" available everywhere without the
-        // need to use "require"
+        // Make the JSX pragma function available everywhere without the need
+        // to use "require"
         new webpack.ProvidePlugin({
-            "E": path.join(__dirname, "element"),
+            [options.jsxPragma]: path.join(__dirname, "element"),
```
Comment (jdan) [inline utils/webpack.config.babel.js]: computed properties :star2:

File: index.js [hunk 2]
```diff
@@ -49,8 +49,7 @@ class Toolbar {
         let $toolbar;
 
         let $defaultPlugins = plugins.default.map((Plugin) => { // eslint-disable-line no-unused-vars
-            // Render each plugin with the bound click handler
-            return <Plugin onClick={::this.handlePluginClick} />;
+            return <Plugin onClick={this.handlePluginClick.bind(this)} />;
```
Comment (rileyjshaw) [inline index.js]: Third option (which I think just won out in our JS style guide) is:

```.js
onClick={() => this.handlePluginClick()}
```

File: index.js [hunk 3]
```diff
@@ -63,7 +62,7 @@ class Toolbar {
                     {
                         plugins.experimental.map((Plugin) => { // eslint-disable-line no-unused-vars
                             return (
-                                <Plugin onClick={::this.handlePluginClick} />
+                                <Plugin onClick={this.handlePluginClick.bind(this)} />
```
Comment (rileyjshaw) [inline index.js]: (ditto)

File: utils/webpack.config.babel.js [hunk 4]
```diff
@@ -1,11 +1,13 @@
-var fs = require("fs");
-var handlebars = require("handlebars");
-var path = require("path");
-var postcss = require("postcss");
-var webpack = require("webpack");
+let fs = require("fs");
```
Comment (rileyjshaw) [inline utils/webpack.config.babel.js]: Can some of these use `const`?

File: utils/webpack.config.babel.js (hunk 1 above)
Comment (rileyjshaw) [inline utils/webpack.config.babel.js]: :star: :star2: :star:

**Comment by rileyjshaw** [comment]
Sweeet!! Lgtm.

Go ahead captain, you have permission to land :rocket:

File: index.js (hunk 2 above)
Comment (jdan) [inline index.js]: This function actually takes an argument, so `() => this.handlePluginClick()` is a no-go :( (without an argument, that is).

This changes in the refactoring I'm doing right now though.

Comment (rileyjshaw) [inline index.js]: Oh! Good point, I hadn't considered that. Hmm...

```
onClick={(...args) => this.handlePluginClick(...args)}
```

That seems a lot worse, right?

Comment (jaredly) [inline index.js]: `onClick={evt => this.handlePluginClick(evt)}` looks fine, right? Also restricts what args might get handed through / depended on

Comment (jdan) [inline index.js]: @jaredly in this specific case the argument is (confusingly) not an event, but I agree that specifically listing out the arguments is very clear and doesn't look bad.

File: utils/webpack.config.babel.js (hunk 4 above)
Comment (jdan) [inline utils/webpack.config.babel.js]: I think I'm going to have one diff in the future that switches all `require`s to use const.

File: index.js (hunk 2 above)
Comment (jdan) [inline index.js]: Our updated styleguide is cool with `.bind` in cases like these - so I am going to keep it. `(...args) => f(...args)` is a little verbose.

**Pull request merged**